### PR TITLE
fix the bug #2213 and #2057, for bug 2057, it will not  appear on Uba…

### PR DESCRIPTION
**Pull request by JoeSun940303**
…ntu VM but will appear on OSX or windows system. We add [read] after rule <k> read and change and change _ to AEs:AExps so that the print can be flushed out both in OSX system and windows system

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/148313234996210/224288619295662)

### CODE REVIEW

File: k-distribution/src/test/java/org/kframework/krun/FlushTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                "    }\n" +
+                "    print(\"Sum = \", s, \"\\n\");\n" +
+                "  }\n" +
+                "}\n";
+        parsed = utils.getParsed(pgm, Source.apply("generated by " + getClass().getSimpleName()));
+    }
+
+
+    @Test
+    public void formatTest() throws IOException {
```
Comment (cos) [inline k-distribution/src/test/java/org/kframework/krun/FlushTest.java:79]: what does this test?

File: k-distribution/src/test/java/org/kframework/kompile/KompileTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    private KRunOptions kRunOptions;
+    private String pgm;
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+
+
+    @Rule
+    public ExpectedException thrown= ExpectedException.none();
+
+    @Test
+    public void KompileTest() throws URISyntaxException, IOException, KEMException {
```
Comment (cos) [inline k-distribution/src/test/java/org/kframework/kompile/KompileTest.java:48]: what does this test? please add javadoc

File: k-distribution/src/test/resources/convertor-tests/PrintFlushTest.k [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                 | "halt" ";"
+                 > "spawn" Stmt
+                 > Stmt Stmt                  [left]
+
+  syntax Ids   ::= List{Id,","}
+  syntax AExps ::= List{AExp,","}
+endmodule
+
+
+module IMP
```
Comment (cos) [inline k-distribution/src/test/resources/convertor-tests/PrintFlushTest.k:33]: Is this the copy-pasted from another place? If so, just refer to that file.

File: k-distribution/tutorial/1_k/4_imp++/lesson_4/imp.k [hunk 4]
```diff
@@ -1,87 +1,93 @@
...
+ syntax Stmt  ::= Block
+                | Id "=" AExp ";"            [strict(2)]
+                | "if" "(" BExp ")"
+                  Block "else" Block         [strict(1)]
+                | "while" "(" BExp ")" Block
+                | "int" Ids ";"
+                | "print" "(" AExps ")" ";"
+                | "halt" ";"
+                > "spawn" Stmt
+                > Stmt Stmt                  [left]
```
Comment (cos) [inline k-distribution/tutorial/1_k/4_imp++/lesson_4/imp.k:26]: please do not reformat the files if not necessary -- it makes reviewing much harder and breaks file change history

**Review by cos — CHANGES_REQUESTED** [review]

File: k-distribution/src/test/java/org/kframework/kompile/KompileTest.java (hunk 2 above)
Comment (JoeSun940303) [reply k-distribution/src/test/java/org/kframework/kompile/KompileTest.java:48]: Sorry about the confusion, just add javadoc to these two tests

File: k-distribution/tutorial/1_k/4_imp++/lesson_4/imp.k (hunk 4 above)
Comment (JoeSun940303) [reply k-distribution/tutorial/1_k/4_imp++/lesson_4/imp.k:26]: Sorry about this, when we tried to fix this bug, we add some spaces to the files but it hard for us to reformat it to the former version. The real change is labeled in dark green in the comparison and all other thing are kept the same.

File: k-distribution/src/test/resources/convertor-tests/PrintFlushTest.k (hunk 3 above)
Comment (JoeSun940303) [reply k-distribution/src/test/resources/convertor-tests/PrintFlushTest.k:33]: This, this is copied from k-distribution/tutorial/1_k/4_imp++/lesson_4/imp.k, the reason we add this file is that if we want to read .k file by using KoreUtils to test it, seems the file should only be placed in resources folder.

File: k-distribution/tutorial/1_k/4_imp++/lesson_4/imp.k (hunk 4 above)
Comment (cos) [reply k-distribution/tutorial/1_k/4_imp++/lesson_4/imp.k:26]: I suggest you redo the changes and on a fresh fork of master. I know how github formatting works, but that doesn't help with breaking the change history. Thanks!

File: k-distribution/src/test/resources/convertor-tests/PrintFlushTest.k (hunk 3 above)
Comment (cos) [reply k-distribution/src/test/resources/convertor-tests/PrintFlushTest.k:33]: In this case, you should minimize the file to whatever is needed to test the fix.

**Comment by daejunpark** [comment]
Close this PR exercise in the class.

**Pull request closed without merging**